posts/2021-09-03-microprofile-21009.adoc

```diff
@@ -0,0 +1,268 @@
+A new method was added to the `ServerEndpointControlMBean`:
+
+`public boolean isActive(String targets);`
+
+The `targets` parameter is a comma separated list of endpoint names. 
+
+This mirrors the existing `isPaused` method. It will return true only if all of the endpoints listed exist, are started, and are not paused.
+
+[source, java]
+----
+// Check if the defaultHttpEndpoint is active
+boolean isEndpointActive = mbean.isActive("defaultHttpEndpoint");
+----
+
+For more information about `ServerEndpointControlMBean` methods, refer to the link:https://www.ibm.com/docs/en/was-liberty/nd?topic=SSAW57_liberty/com.ibm.websphere.javadoc.liberty.doc/com.ibm.websphere.appserver.api.kernel.service_1.0-javadoc/com/ibm/websphere/kernel/server/ServerEndpointControlMBean.html[API documentation].
+
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A21009+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.9].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/16700[Improve featureUtility performance with remote repository]
++
+Previously, the `featureUtility` would take a long time to resolve features when using a remote repository with a slow network. Artifacts would be fetched one at a time which slowed things down dramatically. This issue was fixed by fetching and downloading multiple artifacts at the same time from Maven repository.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18237[Ignore FFDC for IOExceptions in handleMessage]
++
+A bug was discovered where first failure data capture (FFDC) files were incorrectly produced. This would happen when bad JSON data was passed to REST APIs that are using the `jaxrs-2.1` feature with `BYO Jackson`. This issue could occur when a JAX-RS resource or provider throws any `IOException`, and was fixed by preventing FFDC from logging the `IOException`.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18155[JSF faces-config parser throws NPE when namespace missing]
++
+Previous changes made for Jakarta Faces 3.0 added logic to check for the new `https://jakarta.ee/xml/ns/jakartaee` namespace in `faces-config.xml` files. Because the new logic uses `String.contentEquals()`, if a `faces-config.xml` file declares a version but not a namespace, a `NullPointerException` would be thrown during app startup. To fix this issue, the logic has now been changed to use `String.equals()`. Please note that while the NullPointerException has been fixed, valid `faces-config.xml` descriptors should declare valid namespaces, ie. `http://java.sun.com/xml/ns/javaee`, `http://xmlns.jcp.org/xml/ns/javaee`, or `https://jakarta.ee/xml/ns/jakartaee`.
```

Comment:
@mbroz2 pointed out a clarity issue with https://github.com/OpenLiberty/open-liberty/issues/18155 - can you add a line (perhaps a new third sentence) along the lines of `This issue affected all JSF versions on Liberty.` ?